lineage/tHMM.py

```diff
@@ -178,11 +177,16 @@ def fit_list(tHMMobj_list, tolerance=1e-9, max_iter=1000):
 
     # Step 0: initialize with random assignments and do an M step
     # when there are no fixed emissions, we need to randomize the start
+    init_all_gammas = []
     for _, tHMM in enumerate(tHMMobj_list):
         init_gammas = [sp.multinomial.rvs(n=1, p=[1. / tHMM.num_states] * tHMM.num_states, size=len(lineage))
-                       for lineage in tHMM.X]
-
-        do_M_E_step(tHMM, init_gammas)
+                        for lineage in tHMM.X]
+        init_all_gammas.append(init_gammas)
+    if len(tHMMobj_list) > 1:
+        do_M_E_step_atonce(tHMMobj_list, init_all_gammas)
+    else:
+        for i, tHMM in enumerate(tHMMobj_list):
+            do_M_E_step(tHMM, init_all_gammas[i])
```

Comment:
Don't you need to do this for `do_M_step` on line 197, too?

Reply:
Honestly I don't remember why we have both `do_M_step` and `do_M_E_step`... I might be forgetting something critical but it'd be nice if they were combined or one just called the other after some setup.

Reply:
Because the `do_M_step` takes the list of tHMM objects, I wrote this if statement inside it, but the `do_M_E_step` only applies on one tHMM object at a time, so I had to keep the if statement here. But you are right, there is no need to separate them. I will combine the two right away.

Reply:
I removed the `do_M_E_step` but didn't removed the one for `atonce_...`.
Also, it seemed to me that the initialization in the `fit_list()` was not really doing anything and I removed it and built a few of the figures, they were not different at all.
Is it necessary to have random initialization before the first step of EM? I think because we initialize everything in the classes this may not be necessary...

Reply:
This all sounds reasonable. Is this ready for review then?

Reply:
It turned out that by removing the initialization, apparently, it messed with calculating the `lowess` for figure S04. I searched the error (which was this: `FloatingPointError: invalid value encountered in true_divide`) and it doesn't seem to be easily fixable! So brought the `do_M_E_step` back so we could call only that in the initialization part in `fit_list`.